### PR TITLE
[16.0][IMP]account_statement_base:improve search filter

### PR DESCRIPTION
**Pull request by LaetitiaElabore**
improves statement line search  :
- by adding quick access to amount search 
- by allowing amount searches without the exact amount

### CODE REVIEW

**Review by remi-filament — APPROVED** [review]
Great, thanks @LaetitiaElabore 
I think you should remove translations which are manaaged un Weblate

File: account_statement_base/views/account_bank_statement_line.xml [hunk 1]
```diff
@@ -138,14 +138,14 @@
         <search>
             <field
                     name="payment_ref"
-                    filter_domain="['|', '|', ('payment_ref', 'ilike', self), ('ref', 'ilike', self), ('narration', 'ilike', self)]"
-                    string="Label, Ref or Notes"
+                    filter_domain="['|', '|', '|', ('amount', 'ilike', self), ('payment_ref', 'ilike', self), ('ref', 'ilike', self), ('narration', 'ilike', self)]"
```
Comment (pedrobaeza) [inline account_statement_base/views/account_bank_statement_line.xml:141]: Operator `ilike` with a float number seems weird. How is it working?

Comment (LaetitiaElabore) [reply account_statement_base/views/account_bank_statement_line.xml:141]: hi @pedrobaeza 
if I understood well, `ilike` can be use with a float thank to OCB/odoo/osv/operation.py line 1136 by casting to str

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge minor

**Pull request merged**